### DISCUSSION TITLE
How can we use plugin-cloud-storage with registered domain in S3

### DISCUSSION THREAD

**Discussion by sori-ard** [payloadcms/payload#8875] [Q&A]
Currently if we use my-domain name like endpoint is not working. The file is not send to my S3 domain.
The plugin generates URLs like that:

`https://my-bucket.s3.my-endpoint/my-bucket/my-doc.pdf`

We need to have: 
`https://my-bucket.s3.my-domain/my-doc.pdf`

We find a workaround to add domain config:

- Adapter S3 file:

```
const storageS3Adapter = s3Adapter({
    config: {
        endpoint: process.env.S3_ENDPOINT, // https://s3.my-endpoint
        region: process.env.S3_REGION,
        credentials: {
            accessKeyId: process.env.S3_ACCESS_KEY_ID,
            secretAccessKey: process.env.S3_SECRET_ACCESS_KEY,
        },
  },
    bucket: process.env.S3_BUCKET_NAME,
    acl:'public-read'
})

export function adapterS3(args: { collection: CollectionConfig; prefix?: string }) {
    const adapter = storageAdapter(args)

    adapter.generateURL = generateURL

    return adapter
}

function generateURL({ filename, prefix = '' }) {
    const domain = process.env.S3_DOMAIN // https://my-bucket.s3.my-domain

    return `${domain}/${path.posix.join(prefix, filename)}`
}
```

- Use adapter in payload config:

```
plugins: [
    payloadCloud(),
    cloudStorage({
      collections: {
        media: {
          adapter: adapterS3,
          disablePayloadAccessControl: true,
        },
      },
    }),
  ],
```

The file is send to my domain S3 and generated URLs is correct:

`https://my-bucket.s3.my-domain/my-doc.pdf`

There is a better way to use this plugin with registered domain in S3?

**Comment by ortimis** [comment]
I think that in your S3 config you need to set `forcePathStyle: true`.
Maybe this should be noted in the docs (ran into this issue, too)

**Reply by sori-ard** [reply]
I tried but unfortunately is not working.
